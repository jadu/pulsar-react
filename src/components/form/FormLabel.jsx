import React from 'react';
import classnames from 'classnames';

export default class FormLabel extends React.Component {

  static defaultProps = {
    className: 'control__label'
  };

  render() {
    let {
      children,
      className,
      hideLabel,
      htmlFor,
      idGuid,
      required,
      ...props} = this.props;

    let variantClasses = classnames(className, {
      'hide': hideLabel
    });

    let requiredIndicator = required && (
      <span aria-hidden="true" className="required-indicator" data-tippy-content="required">*</span>
    );

    return (
      <>
      {children ? (
        <label 
          htmlFor={htmlFor ? htmlFor : idGuid} 
          className={variantClasses}
        >
          {children}
          {requiredIndicator}
        </label>
      )
      : (
        <span className="control__label"></span>
      )}
      </>
    );
  }
}``