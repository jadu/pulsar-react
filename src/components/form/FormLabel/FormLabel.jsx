import React from 'react';
import classnames from 'classnames';

export default class FormLabel extends React.Component {

  static defaultProps = {
    className: 'control__label',
    tag: 'label'
  };

  render() {
    let {
      children,
      className,
      hideLabel,
      htmlFor,
      idGuid,
      tag,
      required,
      ...props} = this.props;

    let LabelTag = tag;

    let variantClasses = classnames(className, {
      'hide': hideLabel
    });

    let requiredIndicator = required && (
      <span aria-hidden="true" className="required-indicator" data-tippy-content="required">*</span>
    );

    return (
      <React.Fragment>
      {children ? (
        <LabelTag
          htmlFor={htmlFor ? htmlFor : idGuid} 
          className={variantClasses}
          {...props}
        >
          {children}
          {requiredIndicator}
        </LabelTag>
      )
      : (
        <span className="control__label"></span>
      )}
      </React.Fragment>
    );
  }
}