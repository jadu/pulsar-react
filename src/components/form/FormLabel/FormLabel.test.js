'use strict';

import React from 'react';
import { shallow, render } from 'enzyme';

import FormLabel from './FormLabel';

it('should render an empty span if no label is provided', () => {
  const tree = render(
    <FormLabel></FormLabel>
  );
  expect(tree).toMatchSnapshot();
});

it('should render the basic label', () => {
  const tree = render(
    <FormLabel>foo</FormLabel>
  );
  expect(tree).toMatchSnapshot();
});

it('should allow the label to be hidden', () => {
  const tree = render(
    <FormLabel hideLabel="true">foo</FormLabel>
  );
  expect(tree).toMatchSnapshot();
});

it('should add the required indicator', () => {
  const tree = render(
    <FormLabel required="true">foo</FormLabel>
  );
  expect(tree).toMatchSnapshot();
});

it('should allow htmlFor attribute', () => {
  const tree = render(
    <FormLabel htmlFor="bar">foo</FormLabel>
  );
  expect(tree).toMatchSnapshot();
});

it('should allow idGuid to be used as htmlFor attribute', () => {
  const tree = render(
    <FormLabel idGuid="baz">foo</FormLabel>
  );
  expect(tree).toMatchSnapshot();
});

it('should prefer htmlFor over idGuid attribute', () => {
  const tree = render(
    <FormLabel htmlFor="bar" idGuid="baz">foo</FormLabel>
  );
  expect(tree).toMatchSnapshot();
});

it('can use a span instead of a label tag', () => {
  const tree = render(
    <FormLabel tag="span">foo</FormLabel>
  );
  expect(tree).toMatchSnapshot();
});