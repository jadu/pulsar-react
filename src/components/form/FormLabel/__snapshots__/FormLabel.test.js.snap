// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add the required indicator 1`] = `
<label
  class="control__label"
>
  foo
  <span
    aria-hidden="true"
    class="required-indicator"
    data-tippy-content="required"
  >
    *
  </span>
</label>
`;

exports[`should allow htmlFor attribute 1`] = `
<label
  class="control__label"
  for="bar"
>
  foo
</label>
`;

exports[`should allow idGuid to be used as htmlFor attribute 1`] = `
<label
  class="control__label"
  for="baz"
>
  foo
</label>
`;

exports[`should allow the label to be hidden 1`] = `
<label
  class="control__label hide"
>
  foo
</label>
`;

exports[`should prefer htmlFor over idGuid attribute 1`] = `
<label
  class="control__label"
  for="bar"
>
  foo
</label>
`;

exports[`should render an empty span if no label is provided 1`] = `
<span
  class="control__label"
/>
`;

exports[`should render the basic label 1`] = `
<label
  class="control__label"
>
  foo
</label>
`;
